domains/auth/Auth.API/API/Services/OidcProviders/SignaturGruppen.cs

```diff
@@ -47,6 +51,71 @@ public NextStep CreateAuthorizationUri(AuthState state)
         return authorizationUri;
     }
 
+    public async Task<OidcTokenResponse> FetchToken(string code)
+    {
+        var redirectUri = authOptions.ServiceUrl + authOptions.OidcLoginCallbackPath;
+
+        var url = $"{authOptions.OidcUrl}/connect/token";
+
+        var valueBytes = Encoding.UTF8.GetBytes($"{authOptions.OidcClientId}:{authOptions.OidcClientSecret}");
+        var authorization = Convert.ToBase64String(valueBytes);
+
+        var tokenRequest = new HttpRequestMessage(HttpMethod.Post, url);
+        tokenRequest.Headers.Add("Authorization", $"Basic {authorization}");
+        tokenRequest.Content = new FormUrlEncodedContent(new Dictionary<string, string>
+        {
+            { "code", code },
+            { "grant_type", "authorization_code" },
+            { "redirect_uri", redirectUri }
+        });
+
+        var tokenResponse = await httpClient.SendAsync(tokenRequest);
+
+        if (tokenResponse.StatusCode != HttpStatusCode.OK)
+        {
+            logger.LogDebug($"FetchToken: tokenResponse: {tokenResponse.StatusCode}");
+            logger.LogDebug($"connect/token: authorization header: {tokenRequest.Headers}");
+            throw new HttpRequestException(tokenResponse.StatusCode.ToString());
+        }
+
+        var tokenJson = await tokenResponse.Content.ReadAsStringAsync();
+
+        var encoded = JsonSerializer.Deserialize<OidcTokenResponse>(tokenJson);
+
+        return encoded != null ? encoded : throw new FormatException();
+
+        //return encoded != null ? DecodeOidcResponse(encoded) : throw new FormatException();  // FIXME Missing Verification of tokens with "/.well-known/openid-configuration/jwks";
```

Comment:
Is the comment needed?
I think if we have //TODO/FIXME there should be a link to a story fixing it